Moq.AutoMock/MockExtensions.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Moq.Language.Flow;
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace Moq.AutoMock
+{
+    /// <summary>
+    /// TODO
+    /// </summary>
+    public static class MockExtensions
+    {
+        /// <summary>
+        /// Specifies a setup on the mocked type for a call to a void method. 
+        /// All parameters are filled with <see cref ="It.IsAny" /> according to the parameter's type.
+        /// </summary>
+        /// <remarks>
+        /// This may only be used on methods that are not overloaded.
+        /// </remarks>
+        /// <typeparam name="T">Type of the mock</typeparam>
+        /// <param name="mock">The mock</param>
+        /// <param name="methodName">The name of the expected method invocation.</param>
+        /// <returns></returns>
+        public static ISetup<T> SetupWithAny<T>(this Mock<T> mock, string methodName)
+            where T : class
+        {
+            if (mock is null)
+            {
+                throw new ArgumentNullException(nameof(mock));
+            }
+
+            if (methodName is null)
+            {
+                throw new ArgumentNullException(nameof(methodName));
+            }
+
+            LambdaExpression lambdaExpression = GetExpression<T>(methodName);
+
+            MethodInfo setupMethod = mock.GetType().GetMethods()
+                .Single(x => x.Name == nameof(Mock<object>.Setup) && x.ReturnType.GetGenericArguments().Length == 1);
+            return (ISetup<T>)setupMethod.Invoke(mock, new object[] { lambdaExpression })!;
+        }
+
+        /// <summary>
+        /// Specifies a setup on the mocked type for a call to a non-void (value-returning) method. 
+        /// All parameters are filled with <see cref ="It.IsAny" /> according to the parameter's type.
+        /// </summary>
+        /// <remarks>
+        /// This may only be used on methods that are not overloaded.
```

Comment:
I'm glad you documented this, but what's blocking us from supporting this too? It just seems surprising (in a bad way). Overloads are something we tend not to think about, seems best to keep it that way.

Still, supporting this later is also a non-breaking change.

Reply:
Largely laziness on my part. The weird part with the API and multiple overloads is you still have to return an `ISetup` but that instance is tied to a single method setup, not all setups. So supporting all overloads gets to be a weird API on what you return. In addition, [C# 9 added covariant return types](https://docs.microsoft.com/dotnet/csharp/language-reference/proposals/csharp-9.0/covariant-returns) and I did not want to think too hard about what that would mean to return some sort of Setup-like object for multiple methods where the return types could be derived types. 

On the flip side if you don't care to do things like set returns/throw/callback on the `ISetup` return then you really don't need the setup in the first place. A loose mock (which is the default), will give you identical results.